src/main/webapp/script.js

```diff
@@ -90,17 +94,39 @@ function drawRegionsMap(censusDataArray, title) {
 
 // createDataArray takes in the data array returned by the census API
 // and reformats it into a data table for the visualization API.
-function createDataArray(censusDataArray) {
+function createDataArray(censusDataArray, description) {
   const vizDataArray = [];
-  censusDataArray.forEach((state) => {
-    vizDataArray.push([state[0], state[1]/1000]);
-  });
-  // Changes the header of the vizDataArray to match the Visualization API
+  // If the length of the census data array is four, the census call has
+  // generated a percentage and total population so we do a calculation
+  //  to find the % of the total.If the length is not four then we can
+  // just return given population.
+  if (censusDataArray[0].length === 4) {
+    censusDataArray.forEach((state) => {
+      vizDataArray.push([state[0], percentToTotal(state[1], state[2])]);
+    });
+  } else {
+    censusDataArray.forEach((state) => {
+      vizDataArray.push([state[0], state[1]/1000]);
+    });
+  }
+  // Changes the header of the vizDataArray to match Visualization API
   vizDataArray[0][0] = 'State';
-  vizDataArray[0][1] = 'Population';
+  // Add a more accurate descriptor for each state
+  vizDataArray[0][1] = 'Population of people ' +
+    description.age + ' who ' + description.action;
+  // make it moved to
+  if (description.action === 'moved') {
```

Comment:
again this feels hacky. Maybe add a map from action to text to display. if an entry is found for a specific action use the text from the map or use the action as is.

Reply:
Fixed to be hopefully not as hacky